catroid/src/test/java/org/catrobat/catroid/test/content/actions/RepeatActionTest.kt

```diff
@@ -0,0 +1,104 @@
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions
+
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.Script
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.content.StartScript
+import org.catrobat.catroid.content.bricks.ChangeYByNBrick
+import org.catrobat.catroid.content.bricks.FormulaBrick
+import org.catrobat.catroid.content.bricks.RepeatBrick
+import org.catrobat.catroid.content.eventids.EventId
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.FormulaElement
+import org.catrobat.catroid.formulaeditor.Sensors
+import org.catrobat.catroid.test.MockUtil
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class RepeatActionTest(
+    private var iterations: Int,
+    private var deltaY: Int,
+    private var loopCondition: Formula?,
+    private var beforeLoopBrick: FormulaBrick?,
+    private var afterLoopBrick: FormulaBrick?,
+    private var expectedValue: Int
+) {
+    private lateinit var sprite: Sprite
+    private lateinit var project: Project
+    private lateinit var script: Script
+
+    private val delayByContract = 0.020f
+    private val deltaDelayByContract = 0.005f
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{index}")
+        fun parameters() = listOf(
+            arrayOf(4, -10, Formula(4), null, ChangeYByNBrick(50), -40),
+            arrayOf(8, -10, Formula(FormulaElement(FormulaElement.ElementType.SENSOR, Sensors.OBJECT_Y.name, null)),
+                    ChangeYByNBrick(5), ChangeYByNBrick(50), 5),
+            arrayOf(4, -1, Formula(-1), null, ChangeYByNBrick(10), 10),
+            arrayOf(4, -1, Formula(0), null, ChangeYByNBrick(10), 10),
+            arrayOf(4, -1, Formula("3"), null, ChangeYByNBrick(10), 7),
+            arrayOf(4, -1, Formula(Double.NaN), null, ChangeYByNBrick(10), 10))
+    }
```

Comment:
Please remove beforeLoopBrick and afterLoopBrick, those two don't belong here and also falsify this test. Have a look at the original test and the repeat action.